tests/unit/src/test/scala/tests/codeactions/OrganizeImportsLspSuite.scala

```diff
@@ -59,6 +59,36 @@ class OrganizeImportsLspSuite
     scalacOptions = scalacOption
   )
 
+  check(
+    "basic-unsaved",
+    """
+      |package a
+      |import scala.concurrent.duration._
+      |import scala.concurrent.Future<<>>
+      |import scala.concurrent.ExecutionContext.global
```

Comment:
@tgodzik Are you sure that this in this test scalafix is performed over semanticdb file from interactive?

I might be wrong, but I thought that diagnostic is always empty in files from InteractiveSemanticdb. So I think it shouldn't delete `import scala.concurrent.ExecutionContext.global` from imports.

Reply:
yeah, some weird things are happening :/

Reply:
Actually, you were right that it was using the wrong semanticdb, but the semanticdb produced in Metals does contain the right diagnostics if we specify them for the module.

I worked around the issue by creating the unsaved file in tmp so that it can be used by Scalafix. It's not super pretty, but until https://github.com/scalacenter/scalafix/issues/1236 is figured out this should make do. 

There is the mention of a symbol table that might be needed, but I think it's not necessary for organize imports itself.

Reply:
> Actually, you were right that it was using the wrong semanticdb, but the semanticdb produced in Metals does contain the right diagnostics if we specify them for the target.

That's fun. At least I'm glad that the my wrong assumption spotted an issue)
Could you also add a test for a file that has compile-error?

What about the whole approach how we interact with organize-import rule, I thought that at some moment it would be better to extract some part of it's API to be able to execute them without semanticdb file at all. It would allow to make [auto-imports be organized too](https://github.com/scalameta/metals-feature-requests/issues/221).

Reply:
That might make sense, but it's a bigger effort I think.

Reply:
>  but it's a bigger effort I think.

Yep, I agree. I mentioned it only to express that probably this issue in scalafix isn't required for this use-case with organize-imports.

Reply:
> Could you also add a test for a file that has compile-error?

I added a test case for that and also added a check for error diagnostics. We will now fail if the semanticdb was not produced without errors.